lib/carrierwave/uploader/file_size.rb

```diff
@@ -0,0 +1,43 @@
+# encoding: utf-8
+
+module CarrierWave
+  module Uploader
+    module FileSize
+      extend ActiveSupport::Concern
+
+      included do
+        before :cache, :check_size!
+      end
+
+      ##
+      # Override this method in your uploader to provide a Range of Size which
+      # are allowed to be uploaded.
+      # === Returns
+      #
+      # [NilClass, Range] a size range which are permitted to be uploaded
```

Comment:
I realize this is 10 years old but this PR is linked from the changelog, and it's not clearly documented in https://github.com/carrierwaveuploader/carrierwave/blob/master/lib/carrierwave/uploader/file_size.rb so I felt like it would be helpful to comment here.

It feels to me like if it's comparing to `new_file.size` isn't it comparing bytes? When I set a `size_range` the validation messages convert to human friendly units and it does seem like it's using bytes. I set the limit to 1024 and it allowed a file smaller than 1KB and when I tried a 4KB file it rejected it, with the error message "Validation failed: File File size should be less than 1 KB". (It does seem to go for the closest human-friendly unit, so if the limit is in MB or GB it reflects that.) So I think that the value here is in fact interpreted as bytes.